src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -119,3 +533,1006 @@ async fn test_decider_installed() {
+            };
+            send_response.send(Ok(response)).unwrap();
+        }
+        assert_eq!(
+            register_worker_counter, 2,
+            "expect try register 2 workers for each deal"
+        );
+    }
+
+    wait_decider(&mut client).await;
+
+    let (last_seen, deals, mut workers) = {
+        let mut result = execute(
+            &mut client,
+            r#"
+            (seq
+                (call relay ("decider" "get_string") ["last_seen_block"] last_seen)
+                (seq
+                    (call relay ("decider" "list_get_strings") ["joined_deals"] deals)
+                    (call relay ("worker" "list") [] workers)
+                )
+            )
+        "#,
+            "last_seen deals workers",
+            hashmap! {},
+        )
+        .await
+        .unwrap();
+        let last_seen = serde_json::from_value::<StringValue>(result.remove(0)).unwrap();
+
+        let deals = serde_json::from_value::<StringListValue>(result.remove(0)).unwrap();
+        assert_eq!(deals.strings.len(), 2);
+        let deals = deals
+            .strings
+            .iter()
+            .map(|deal| serde_json::from_str::<JoinedDeal>(deal).unwrap())
+            .collect::<Vec<_>>();
+
+        let workers = serde_json::from_value::<Vec<String>>(result.remove(0))
+            .unwrap()
+            .into_iter()
+            .collect::<HashSet<_>>();
+
+        (last_seen, deals, workers)
+    };
+    // Note that it must not be BLOCK_NUMBER_2 since we save BLOCK_NUMBER_2 - 1
+    assert_eq!(last_seen.str, to_hex(BLOCK_NUMBER_1));
+
+    let mut expected = hashmap! {
+        deal_id_1 => (TestApp::test_app1(), BLOCK_NUMBER_1),
+        deal_id_2 => (TestApp::test_app2(), BLOCK_NUMBER_2),
+    };
+    for deal in deals {
+        let worker_exist = workers.remove(&deal.worker_id);
+        assert!(
+            worker_exist,
+            "worker_id from joined_deals is not in the list of peer workers"
+        );
+        let result = expected.remove(&deal.deal_id);
+        assert!(
+            result.is_some(),
+            "deal_id from joined_deals is not in the list of expected deals"
+        );
+        let (app, block) = result.unwrap();
+
+        let cid = get_worker_app_cid(&mut client, &deal.worker_id).await;
+        assert_eq!(cid, app.cid, "wrong cid");
+        let deal_state = get_deal_state(&mut client, &deal.deal_id).await;
+        assert_eq!(deal_state.left_boundary, to_hex(block), "wrong saved state");
+    }
+
+    server.shutdown().await;
+}
+
+/// Test deal deployment in different *Decider* runs
+///
+/// Plan
+/// 1. *Decider* installs a deal.
+///     This process already was checked.
+/// 2. *Decider* installs another deal.
+///    Check exactly what can be changed from the installation of another deal
+///    a. `joined_deals` list contains both deals (check that the list is not overwritten)
+///    b. state of the deal (stored by `deal_id`)
+/// 3. Install a
```

Comment:
seems like part of the comment was deleted :(

Reply:
don't remember what I wanted to write here :see_no_evil: 
edited